contributed_definitions/nyaml/NXmpes.yaml

```diff
@@ -230,58 +232,139 @@ NXmpes(NXobject):
             This is the momentum axis to be used for data plotting.
     (NXsample):
       name:
-      chemical_formula:
-        exists: recommended
-        doc: |
-          The chemical formula of the sample. For mixtures use the NXsample_component
-          group in NXsample instead.
-      sample_history(NXnote):
-        exists: recommended
+      (NXsubstance):
         doc: |
-          A descriptor to keep track of the treatment of the sample before entering the
-          photoemission experiment. Ideally, a full report of the previous operations, in
-          any format (NXnote allows to add pictures, audio, movies). Alternatively, a
-          reference to the location or a unique identifier or other metadata file. In the
-          case these are not available, free-text description.
+          For samples containing a single pure substances. For mixtures use the
+          NXsample_component_set and NXsample_component group in NXsample instead.
+        molecular_formula_hill:
+          exists: recommended
+          doc: |
+            The chemical formula of the sample.
       atom_types:
         exists: recommended
         doc: |
           List of comma-separated elements from the periodic table
           that are contained in the sample.
           If the sample substance has multiple components, all
           elements from each component must be included in `atom_types`.
-      preparation_date(NX_DATE_TIME):
+      situation:
+        exists: optional
+        enumeration: [vacuum, inert atmosphere, oxidising atmosphere, reducing atmosphere]
+      (NXsample_history):
         exists: recommended
         doc: |
-          Date of preparation of the sample for the XPS experiment (i.e. cleaving, last
-          annealing).
-      preparation_description(NXnote):
-        exists: optional
+          A set of activities that occurred to the sample prior to/during photoemission experiment.
+          
+          Ideally, a full report of the previous operations (or links to a chain of operations).
+          Alternatively, notes allow for additional descriptors in any format
+          (NXnote allows to add pictures, audio, movies). Alternatively, a
+          reference to the location or a unique identifier or other metadata file. In the
+          case these are not available, free-text description.
+        sample_preparation(NXphysical_process):
+          exists: recommended
+          doc: |
+            Details about the sample preparation for the MPES experiment (i.e. cleaving, last
+            annealing).
+          start_time(NX_DATE_TIME):
+          end_time(NX_DATE_TIME):
+            exists: recommended
+          method:
+            exists: optional
+            doc: |
+              Details about the method of sample preparation before the MPES experiment.
+          notes:
+            exists: optional
+      temperature(NXenvironment):
```

Comment:
Ok, now I understand your point with the temperature. Yes we should change the docstring in NXmanipulator, that it is the same concept and value as NXsample/thermocouple/value

Reply:
> Yes, this is correct and it also works like it is. My point is that it would be nice to store these parameters as an NXenvironment on the manipulator side as well. This would allow us to have the description of the NXsensors measuring the temperature, pressure, and so on, within NXinstrument (where they should arguably belong).

Yes, now I understood what you mean. I like the suggestion to add and link the NXsensor in the manipulator as this fits in the instrument part of the experiment.